packages/@aws-cdk/aws-stepfunctions/lib/state-machine.ts

```diff
@@ -297,12 +304,24 @@ export class StateMachine extends StateMachineBase {
       }));
     }
 
+    let tracingConfiguration: CfnStateMachine.TracingConfigurationProperty | undefined;
+    if (props.tracingEnabled) {
+      tracingConfiguration = {
+        enabled: true,
+      };
+      this.addToRolePolicy(new iam.PolicyStatement({
+        actions: ['xray:PutTraceSegments', 'xray:PutTelemetryRecords'],
+        resources: ['*'],
+      }));
+    }
+
     const resource = new CfnStateMachine(this, 'Resource', {
       stateMachineName: this.physicalName,
       stateMachineType: props.stateMachineType ? props.stateMachineType : undefined,
       roleArn: this.role.roleArn,
       definitionString: Stack.of(this).toJsonString(graph.toGraphJson()),
       loggingConfiguration,
+      tracingConfiguration,
```

Comment:
assuming you've moved it to a `setupXray()` method, you would then do

```suggestion
      tracingConfiguration: props.tracingEnabled ? this.setupXray() : undefined,
```

Reply:
went with a slightly different approach - similar to https://github.com/aws/aws-cdk/blob/a5e77e47d3bc7a326be56eef527db6f988b8ac9f/packages/%40aws-cdk/aws-lambda/lib/function.ts#L906